### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by clangenb**
Added minimal integration tests. I made them primarily to showcase the usage of the api a bit more.

Note:
* first merge: #15

### CODE REVIEW

**Comment by clangenb** [comment]
Converted to draft. As I still need to tweak CI and add more tests afterwards.

**Comment by clangenb** [comment]
@anizeani I added here some integration tests that test the custom rpcs. It seems to work here. I don't know what your problem is.

You can run these tests locally the same way they are run in the CI to tests.

File: packages/node-api/src/e2e.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            // console.log(result);
+            expect(result.length).toBe(0);
+        });
+
+        it('communities.GetOfferingsForBusiness should return empty vec', async () => {
+            // default bid
+            const cid = '4SkU25tusVChcrUprW8X22QoEgamCgj3HKQeje7j8Z4E'
+            const alice = keyring.addFromUri('//Alice', { name: 'Alice default' })
+
+            const bid= api.createType('BusinessIdentifier', [cid, alice.publicKey]);
```
Comment (clangenb) [inline packages/node-api/src/e2e.ts:59]: @anizeani sorry for the late example. This is how you'd do it. You could also pass the cid/pubkey in a json. This pattern applies for all types that are defined as follows in rust:

https://github.com/encointer/pallets/blob/5852ace4431df10f20507c7509d8cdd03f6bf2d4/primitives/src/bazaar.rs#L27

File: .github/workflows/integration_test.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Download encointer-node
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{secrets.GITHUB_TOKEN}}
+          workflow: build-and-test.yml
+          name: executables
+          # in fact this action should download the latest artifact, but sometimes fails. Then we need to
+          # set the `run_id` to force a download of an updated binary.
+          run_id: 1282695776
```
Comment (brenzi) [inline .github/workflows/integration_test.yml:28]: artifacts arw not persisted for very long. we can't rely on runs. can we use the latest release instead?

Comment (brenzi) [reply .github/workflows/integration_test.yml:28]: We can leave it until we run into issues ;-) It's done in other repos too anyway

**Pull request merged**